single-line-diagram/single-line-diagram-core/src/main/java/com/powsybl/sld/SingleLineDiagram.java

```diff
@@ -97,74 +79,46 @@ public static void drawVoltageLevel(Network network, String id, String svgFile)
-    }
-
-    public static void drawSubstation(Network network, String id, Path svgFile, LayoutParameters layoutParameters) {
-        drawSubstation(network, id, svgFile, layoutParameters, new ConvergenceComponentLibrary());
+        drawSubstation(network, id, svgFile, new ParamBuilder().build());
     }
 
-    public static void drawSubstation(Network network, String id, Path svgFile, LayoutParameters layoutParameters, ComponentLibrary componentLibrary) {
-        drawSubstation(network, id, svgFile, layoutParameters, componentLibrary,
-                new DefaultDiagramLabelProvider(network, componentLibrary, layoutParameters),
-                createDefaultStyleProvider(network),
-                "");
-    }
+    private static void drawSubstation(Network network, String substationId, Path svgFile, Param param) {
 
-    public static void drawSubstation(Network network, String id, Path svgFile, LayoutParameters layoutParameters, ComponentLibrary componentLibrary,
-                                      DiagramLabelProvider initProvider, StyleProvider styleProvider, String prefixId) {
-        drawSubstation(network, id, svgFile, layoutParameters, componentLibrary,
-                        new HorizontalSubstationLayoutFactory(), new SmartVoltageLevelLayoutFactory(network),
-                        initProvider, styleProvider, prefixId);
-    }
+        LayoutParameters layoutParameters = param.getLayoutParameters();
+        VoltageLevelLayoutFactory voltageLevelLayoutFactory = param.getVoltageLevelLayoutFactory();
+        SubstationLayoutFactory substationLayoutFactory = param.getSubstationLayoutFactory();
 
-    private static void drawSubstation(Network network, String substationId, Path svgFile, LayoutParameters layoutParameters, ComponentLibrary componentLibrary,
-                                       SubstationLayoutFactory sLayoutFactory, VoltageLevelLayoutFactory vLayoutFactory,
-                                       DiagramLabelProvider initProvider, StyleProvider styleProvider, String prefixId) {
-        Objects.requireNonNull(sLayoutFactory);
-        Objects.requireNonNull(vLayoutFactory);
+        Objects.requireNonNull(substationLayoutFactory);
+        Objects.requireNonNull(voltageLevelLayoutFactory);
 
         SubstationGraph substationGraph = new NetworkGraphBuilder(network).buildSubstationGraph(substationId);
-        sLayoutFactory.create(substationGraph, vLayoutFactory).run(layoutParameters);
-        draw(substationGraph, svgFile, layoutParameters, componentLibrary, initProvider, styleProvider, prefixId);
+        substationLayoutFactory.create(substationGraph, voltageLevelLayoutFactory).run(layoutParameters);
+        DefaultSVGWriter svgWriter = new DefaultSVGWriter(param.getComponentLibrary(), param.getLayoutParameters(), param.getSvgParameters());
+        draw(substationGraph, svgFile, svgWriter, param.createLabelProvider(network), param.createStyleProvider(network), param.getSvgParameters().getPrefixId());
```

Comment:
I removed this prefeixId parameter from several subfunctions in the DefaultSvgWriter as well, since it is contained in the metadata (already a parameter of those subfunctions) and that calling it from the metadata is quite simple (simple getters).